[inductor][heuristics] add kernel template params

coconutruben opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #163206
* #163209
* __->__ #162781

# why

- enable a clear interface for kernel templates to declare all their
  instantiation parameters and any potential defaults
- simplify KernelTemplateChoice to just have a single params, and not kwargs and extra_kwargs

# what

- KernelTemplateParams interface
- placeholder implementation where we just pass through a dict

# testing

- existing ci tests

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

coconutruben committed:
[inductor][heuristics] add kernel template params

\# why

- enable a clear interface for kernel templates to declare all their
  instantiation parameters and any potential defaults

\# what

- KernelTemplateParams interface
- placeholder implementation where we just pass through a dict

\# testing

- existing ci tests

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][heuristics] add kernel template params"


# why

- enable a clear interface for kernel templates to declare all their
  instantiation parameters and any potential defaults
- simplify KernelTemplateChoice to just have a single params, and not kwargs and extra_kwargs

# what

- KernelTemplateParams interface
- placeholder implementation where we just pass through a dict

# testing

- existing ci tests

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

jansel reviewed (changes requested):
When will this be something other than a dict?  Not totally clear why we need to wrap the dict in a class...

jansel commented on torch/_inductor/template_heuristics/params.py:
```diff
@@ -0,0 +1,42 @@
+    def to_kwargs(self) -> dict[str, Any]:
+        return self.kwargs.copy()
+
+    def to_serializeable_dict(self) -> dict[str, Any]:
```
When would `to_kwargs` and `to_serializeable_dict` be different?

coconutruben commented:
@pytorchbot merge

coconutruben committed:
Update on "[inductor][heuristics] add kernel template params"


# why

- enable a clear interface for kernel templates to declare all their
  instantiation parameters and any potential defaults
- simplify KernelTemplateChoice to just have a single params, and not kwargs and extra_kwargs

# what

- KernelTemplateParams interface
- placeholder implementation where we just pass through a dict

# testing

- existing ci tests

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][heuristics] add kernel template params"


# why

- enable a clear interface for kernel templates to declare all their
  instantiation parameters and any potential defaults
- simplify KernelTemplateChoice to just have a single params, and not kwargs and extra_kwargs

# what

- KernelTemplateParams interface
- placeholder implementation where we just pass through a dict

# testing

- existing ci tests

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben commented:
@pytorchbot merge
